workspace/workspace.go

```diff
@@ -216,15 +226,68 @@ func LoadWorkspace() (Workspace, error) {
 	manifestFile := filepath.Join(workspacePath, ".yourbase.yml")
 	configFile := filepath.Join(workspacePath, "config.yml")
 
+	var w Workspace
+
 	if PathExists(manifestFile) {
 		log.Debugf("Loading workspace from manifest file: %s", manifestFile)
-		return loadWorkspaceFromPackage(manifestFile, workspacePath)
+		w, err = loadWorkspaceFromPackage(manifestFile, workspacePath)
+	}
+
+	if err == nil && validWorkspaceConfig(w) {
+		return w, nil
 	}
 
 	if PathExists(configFile) {
 		log.Debugf("Loading workspace from config file: %s", configFile)
-		return loadWorkspaceFromConfigYaml(configFile, workspacePath)
+		w, err = loadWorkspaceFromConfigYaml(configFile, workspacePath)
+	}
+
+	if err == nil && validWorkspaceConfig(w) {
+		return w, nil
+	}
+
+	return Workspace{}, fmt.Errorf("Error resolving workspace at path %s", workspacePath)
+}
+
+// validWorkspaceConfig will check if every expected part of whole Configuration Graph is initialized
+// as expected
+func validWorkspaceConfig(workspace Workspace) (valid bool) {
```

Comment:
Consider returning an error (or multiple) from here to help the user in determining what part fails validation.

Reply:
Next PR, got it! :wink: